currentgk.py

```diff
@@ -1,391 +1,255 @@
-                        quizname = kotobadict["sessionName"]
-                    else:
-                        quizname = ""
-                        for deckdict in kotobadict["decks"]:
-                            addname = deckdict["name"]
-                            quizname += " " + addname
-
-                    startindex = 0
-                    endindex = 0
-
-                    mc = kotobadict["decks"][0]["mc"]
-
-                    try:
-                        startindex = kotobadict["decks"][0]["startIndex"]
-                        endindex = kotobadict["decks"][0]["endIndex"]
-
-                    except KeyError:
-                        pass
-
-                    anticheatinfo = startindex, endindex, mc, shuffle, isloaded
-
-                    print(f"Score limit: {scorelimit}\nAnswer time limit: {answertimelimitinms}"
-                          f"\nFont size: {fontsize}\nFont: {font}\nQuiz name: {quizname}"
-                          f"\nUser count: {usercount}\nUser id: {mainuserid}\nQuestion count: {questioncount}"
-                          f"\nFailed question count: {failedquestioncount}\nScored points: {myscore}")
-
-                    quizinfo = (scorelimit, answertimelimitinms, fontsize, font, quizname, usercount, mainuserid,
-                                failedquestioncount, myscore, anticheatinfo, quizcommand)
-                    return quizinfo
-
-
-# Ready Message / Bot Config
-@meido.event
-async def on_ready():
-    # await meido.change_presence(activity=discord.Game('.levelup for commands'))
-    print(f"Logged in as\n\tName: {meido.user.name}\n\tID: {meido.user.id}")
-    
-    #TIMM: setting up quiz_tries.db
-    _set_globals()
-    print(f'Initing tables on {_DB_NAME}')
-    global store
-    store = Store(_DB_NAME)
-    init_tables(_DB_NAME)
-    print('Done initing tables')
-    
-
-#TIMM: adding mechanism for storing quiz tries
-async def fail_message(message, mainuserid, quizcommand, created_at, result, myguild):
-    if quizcommand == "k!quiz jpdb1k(1-300) 25 hardcore nd mmq=10 dauq=1 font=5 color=#f173ff size=100":
+def get_roles(guild):
+    roles = [f for k in RANK_NAMES if (f := get(guild.roles, name=k))]
+    if len(roles) != len(RANK_NAMES):
         return
-    store.new_quiz_attempt(mainuserid, quizcommand, created_at, result)
-    unixstamp = store.get_unix()
-    user = myguild.get_member(mainuserid)
-    await message.channel.send(f"Please attempt again in <t:{int(unixstamp[0])}:R> at <t:{unixstamp[0]}>. Any attempts until then will not be counted.")
-    await user.send(f'Please attempt again in <t:{int(unixstamp[0])}:R> at <t:{unixstamp[0]}>. Any attempts on ``{quizcommand}`` until then will not be counted.')
+    return dict(zip(RANK_NAMES, roles))
 
-#TIMM: cooldown message after failed quiz    
-async def cooldown_message(message, mainuserid, quizcommand, logs, created_at, myguild):
+
+async def fail(store, quiz, guild, member_id):
```

Comment:
im changing the guild to message since the await message.channel.send doesnt work otherwise

Reply:
Oh I think I forgot to pass in the channel